maven-wrapper-plugin/src/main/java/org/apache/maven/plugins/wrapper/WrapperMojo.java

```diff
@@ -405,4 +519,184 @@ protected String determineRepoUrl(String envRepoUrl, Settings settings) {
 
         return DEFAULT_REPOURL;
     }
+
+    private void showJdkVersionWarnings() {
+        boolean warningsShown = false;
+
+        // Check main JDK version
+        if (jdkVersion != null && jdkDistributionUrl == null) {
+            if (checkJdkVersionStability("Main JDK", jdkVersion, jdkDistribution)) {
+                warningsShown = true;
+            }
+        }
+
+        // Check toolchain JDK version
+        if (toolchainJdkVersion != null && toolchainJdkDistributionUrl == null) {
+            if (checkJdkVersionStability("Toolchain JDK", toolchainJdkVersion, toolchainJdkDistribution)) {
+                warningsShown = true;
+            }
+        }
+
+        // Only show suppression message if warnings were actually displayed
+        if (warningsShown) {
+            getLog().warn("To suppress these warnings, use -DskipJdkWarnings=true");
+            getLog().warn("");
+        }
+    }
+
+    private boolean checkJdkVersionStability(String jdkType, String version, String distribution) {
+        if (version == null) {
+            return false;
+        }
+
+        boolean warningShown = false;
+
+        // Check for non-LTS versions
+        if (isNonLtsVersion(version)) {
```

Comment:
I assume you also want this for Maven 3? Otherwise maybe use multiline string

Reply:
Not sure what you mean here ?

Reply:
You log a lot of single line strings to generate something that looks like a multiline string? Since java 13 (?) Java supports texts blocks like. As we set Java 17 for Maven 4, maybe we can make use of this here if the change is only aiming at Maven 4.

Reply:
Ah, got it.  So this is currently not focused on Maven 4. Also, using multiline strings changes the output, as only the first line will be prepended with `[WARN] ` afaik, so it kinda messes the output which is not correctly aligned anymore.